### PR TITLE
Dolphin save state symlink fix

### PR DESCRIPTION
**Pull request by DylanTackoor**
Attempts to resolve https://github.com/dragoonDorise/EmuDeck/issues/355 by linking to the `StateSaves` folder for new installs and by migrating existing files from the old `states` folder to the new one. Wasn't sure if the migrate function was a good place to put this logic in, can move it elsewhere if needed 🤷

### CODE REVIEW

**Comment by DylanTackoor** [comment]
I thought it would be cool to set Dolphin's `StateSave` directory directly, but I could only find forum threads about [it not being configurable ](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/FileUtil.cpp#L959) :(

**Comment by Godsbane** [comment]
what do you think about changing from saving in the .var folder to moving the save to the Emulation folder, and then using the moveSaveFolder function to move the files and link them back to the spot it's looking for?

**Comment by DylanTackoor** [comment]
I initially left it under `.var` just because the `GC` and `Wii` folders were also symlinks pointing to `.var`, but yeah having the files stored under the Emulation folder feels nice. I think I've `moveSaveFolder`-ed both the folders correctly now 🤔

File: functions/helperFunctions.sh [hunk 1]
```diff
@@ -444,4 +444,21 @@ function createDesktopShortcut(){
...
+	local oldStatesPath="$HOME/.var/app/$flatpakName/data/dolphin-emu/states"
+	local newStatesPath="$HOME/.var/app/$flatpakName/data/dolphin-emu/StateSaves"
+	local linkedTarget=$(readlink -f "$savesPath/$emuName/states")
+
+	if [[ -e "$oldStatesPath" && "$oldStatesPath" == "$linkedTarget" ]]; then
+		echo "Migrating $emuName states from $oldStatesPath to $newStatesPath"
+		cp -r "$oldStatesPath/*" "$newStatesPath"
+		rm -rf "$oldStatesPath"
+		unlink "$savesPath/$emuName/states"
+		moveSaveFolder dolphin states "$HOME/.var/app/$flatpakName/data/dolphin-emu/StateSaves"
```
Comment (Godsbane) [inline functions/helperFunctions.sh:462]: this will not work for primehack, because it will try to link it to the Emulation/dolphin/states folder

**Review by Godsbane — CHANGES_REQUESTED** [review]
Please look at the migrateDolphinStates function, and make sure it works for dolphin derivatives as well by using the emuName variable.

**Comment by DylanTackoor** [comment]
hah I just realized my text editor formatted a file even though I was trying to be careful about it with this PR 🤦  I'll try to revert those formatting changes later tonight tomorrow if they're annoying

**Review by Godsbane — APPROVED** [review]
To me this looks good.

I don't see the point of migrating states to SaveStates, because (as far as I know) there's never actually anything in states, and the link was just created to the wrong spot. That said, i like the function and think it should stay just in case.

also like the new helper function. I'm going to look at extending it a bit. I'll commit this PR to dev and we can work shop / test from there.

**Pull request merged**